chainerx_cc/chainerx/cuda/cuda_device/batch_norm.cc

```diff
@@ -239,10 +251,10 @@ class CudaBatchNormGradKernel : public BatchNormGradKernel {
 
         Array gout_cont = AsContiguous(gout);
         Array actual_gx = EmptyLike(x, device);
-        cuda_internal::CudnnTensorDescriptor x_desc{x_cont};
+        cuda_internal::CudnnTensorDescriptor x_desc{ExpandToAtLeast4D(x_cont)};
 
         // The CudnnTensorDescriptor for `x_cont` can be reused for `gout_cont`.
-        CHAINERX_ASSERT(x_desc.GetDtype() == cuda_internal::CudnnTensorDescriptor{gout_cont}.GetDtype());
+        CHAINERX_ASSERT(x_cont.dtype() == gout_cont.dtype());
```

Comment:
Let's write as `CHAINERX_ASSERT(x.dtype() == gout.dtype())` and move to the top of the function.